tls/s2n_key_update.c

```diff
@@ -18,26 +18,83 @@
 #include "tls/s2n_connection.h"
 #include "tls/s2n_key_update.h"
 #include "tls/s2n_tls13_handshake.h"
+#include "tls/s2n_record.h"
 
 #include "utils/s2n_safety.h"
 
-const uint8_t S2N_KEY_UPDATE_NOT_REQUESTED = 0;
-const uint8_t S2N_KEY_UPDATE_REQUESTED = 1;
+int s2n_key_update_write(struct s2n_blob *out);
 
-int s2n_key_update_recv(struct s2n_connection *conn)
+int s2n_key_update_recv(struct s2n_connection *conn, struct s2n_stuffer *request)
 {
     notnull_check(conn);
+
     uint8_t key_update_request;
-    GUARD(s2n_stuffer_read_uint8(&conn->handshake.io, &key_update_request));
+    GUARD(s2n_stuffer_read_uint8(request, &key_update_request));
     S2N_ERROR_IF(key_update_request != S2N_KEY_UPDATE_NOT_REQUESTED && key_update_request != S2N_KEY_UPDATE_REQUESTED,
             S2N_ERR_BAD_MESSAGE);
     conn->key_update_pending = key_update_request;
+
     /* Update peer's key since a key_update was received */
     if (conn->mode == S2N_CLIENT){
         GUARD(s2n_update_application_traffic_keys(conn, S2N_SERVER, RECEIVING));
     } else {
         GUARD(s2n_update_application_traffic_keys(conn, S2N_CLIENT, RECEIVING));
     }
 
-    return 0;
+    return S2N_SUCCESS;
+}
+
+int s2n_key_update_send(struct s2n_connection *conn, size_t size) 
```

Comment:
Should new functions be returning S2N_RESULT instead of int?

Reply:
I think it's probably best to stay consistent with the module

Reply:
I started this code change before we had S2N_RESULT, this will have to get migrated over as well.